### ISSUE TITLE
Performance regression with 1.1.10.4?

### ISSUE THREAD

**Issue by whenamanlies** [xerial/snappy-java#542]
We are moving from 1.1.10.3 to 1.1.10.4 (for CVE-2023-43642) and observing performance regressions (reading parquet, if that's important). Anyone else observing similar?

Wonder if there's a tie to snappy's upgrade from 1.1.8 to 1.1.10.
Snappy project has been quiet lately however there's at least [one similar report](https://groups.google.com/g/snappy-compression/c/eEdVnfggWak/m/_pC6CH-MAgAJ) in snappy-compression Google Group.

**Comment by xerial** [comment]
Since https://github.com/xerial/snappy-java/releases/tag/v1.1.10.4, the internal snappy version has upgraded to snappy 1.1.10. Until snappy-java 1.1.10.3, snappy 1.1.8 was unexpectedly used. 

If the original Snapy 1.1.10 has performance regression, it should be the cause. The CVE fix would be unrelated as Parque is not using SnappyInput/OutputStream as far as I know.

**Comment by xerial** [comment]
@whenamanlies No. The best way is to report the regression at snappy https://github.com/google/snappy and fix it there.

**Comment by findmenowhere** [comment]
Hello @whenamanlies, we recently had the same issue, and the Google Group has been banned. Have you reached any conclusions?

**Comment by findmenowhere** [comment]
@xerial Also do we mind upgrading the internal snappy version to 1.2.x in order to verify if the regression is already fixed in the newer version?
